File: src/test/java/com/github/loadtest4j/drivers/gatling/GatlingFactoryTest.java
package com.github.loadtest4j.drivers.gatling;

import com.github.loadtest4j.loadtest4j.driver.Driver;
import com.github.loadtest4j.loadtest4j.driver.DriverFactory;
import org.junit.Test;

import java.util.Map;
import java.util.Set;
import java.util.concurrent.ConcurrentHashMap;

import static org.assertj.core.api.Assertions.assertThat;

public class GatlingFactoryTest {
    @Test
    public void testGetMandatoryProperties() {
        final DriverFactory sut = driverFactory();

        final Set<String> mandatoryProperties = sut.getMandatoryProperties();

        assertThat(mandatoryProperties)
                .containsExactly("duration", "url");
    }

    @Test(expected = UnsupportedOperationException.class)
    public void testGetMandatoryPropertiesIsImmutable() {
        final DriverFactory sut = driverFactory();

        sut.getMandatoryProperties().add("foobarbaz123");
    }

    @Test
    public void testCreate() {
        final DriverFactory sut = driverFactory();

        final Map<String, String> properties = new ConcurrentHashMap<>();
        properties.put("duration", "2");
        properties.put("url", "https://example.com");

        final Driver driver = sut.create(properties);

        assertThat(driver)
                .isNotNull();
    }

    private DriverFactory driverFactory() {
        return new GatlingFactory();
    }
}

File: src/test/scala/com/github/loadtest4j/drivers/gatling/GatlingFactoryTest.scala
package com.github.loadtest4j.drivers.gatling

import com.github.loadtest4j.loadtest4j.DriverFactory
import org.junit.Test
import org.junit.Assert._

import scala.collection.JavaConverters

class GatlingFactoryTest {
    @Test
    def testGetMandatoryProperties(): Unit = {
        val sut = driverFactory()

        val mandatoryProperties = sut.getMandatoryProperties

        assertEquals(2, mandatoryProperties.size())
        assertTrue(mandatoryProperties.contains("duration"))
        assertTrue(mandatoryProperties.contains("url"))
    }

    @Test
    def testCreate(): Unit = {
        val sut = driverFactory()

        val properties = JavaConverters.mapAsJavaMap(Map("duration" -> "2", "url" -> "https://example.com"))

        val driver = sut.create(properties)

        assertNotNull(driver)
    }

    private def driverFactory(): DriverFactory = {
        new GatlingFactory()
    }
}
